crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -960,6 +964,26 @@ mod serde_impl {
                 .ok_or(String::from("Expected hex string to be prefixed by '0x'"))
                 .map_err(de::Error::custom)
         }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: de::SeqAccess<'de>,
+        {
+            let mut bytes = [0u8; 32];
+            for (i, byte) in bytes.iter_mut().enumerate() {
+                *byte = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(i, &"32 bytes"))?;
```

Comment:
For extra compactness, can't we accept sequences with fewer elements than 32?
Eg: Felt::ZERO could be serialized as `[0]`, rather than `[0; 32]`.
And all values under 255 on a single element, and so on.

Reply:
We are kinda doing this already with the human-readable form. Felt::ZERO is `0x0` not `0x000000000..0000`

Reply:
Oh I can see @LucasLvy already suggested this.
I agree with him